packages/tenant-process/src/model/domain/errors.ts

```diff
@@ -314,3 +316,22 @@ export function tenantHasNoDelegatedProducerFeature(
     title: "Feature not assigned",
   });
 }
+
+export function agreementNotFound(agreementId: string): ApiError<ErrorCodes> {
+  return new ApiError({
+    detail: `Agreement ${agreementId} not found`,
+    code: "agreementNotFound",
+    title: "Agreement not found",
+  });
+}
+
+export function descriptorNotFoundInEservice(
+  descriptorId: string,
+  eservice: string
```

Comment:
eservice is more generic, it's more clear to use `eserviceId`
```suggestion
  eserviceId: string
```